share/eds/rsmt2d.go

```diff
@@ -46,8 +46,12 @@ func (eds *Rsmt2D) AxisRoots(context.Context) (*share.AxisRoots, error) {
 // Sample returns share and corresponding proof for row and column indices.
 func (eds *Rsmt2D) Sample(
 	_ context.Context,
-	rowIdx, colIdx int,
+	idx shwap.SampleIndex,
 ) (shwap.Sample, error) {
+	rowIdx, colIdx, err := idx.Coordinates(shwap.EdsIDSize)
+	if err != nil {
+		return shwap.Sample{}, err
+	}
 	return eds.SampleForProofAxis(rowIdx, colIdx, rsmt2d.Row)
```

Comment:
We should use the SampleIndex here as well. The objective is to use a single type everywhere

Reply:
Ok, `SampleForProofAxis` is now:

```
func (eds *Rsmt2D) SampleForProofAxis(
	idx shwap.SampleIndex,
	proofType rsmt2d.Axis,
) (shwap.Sample, error) {
```